include/nuttx/crc8.h

```diff
@@ -28,6 +28,14 @@
 #include <sys/types.h>
 #include <stdint.h>
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Append full suffix to avoid the penitential symbol collision */
+
+#define crc8   crc8full
```

Comment:
1. why not rename function instead?
2. why not resolve collision at third party side?
3. is `#define crc8ccitt crc8ccittfull` needed?

Reply:
> 1. why not rename function instead?

To avoid modify the caller

> 2. why not resolve collision at third party side?

It's hard to modify other widely used library(e.g. zlib: https://github.com/madler/zlib/blob/master/zlib.h#L1727)

> 3. is `#define crc8ccitt crc8ccittfull` needed?

The possibility of conflict is small for crc8ccitt.